### PR TITLE
Add limiting for 30x redirects

### PR DESCRIPTION
**Pull request by jrozner**
Implementation of issue #49 for limiting the maximum number of redirects allowed to follow.

### CODE REVIEW

**Comment by davecheney** [comment]
The test failure is my fault from merging #34. Please merge master and push again.

**Comment by davecheney** [comment]
I'm strongly opposed to making this a configurable constant. The users of this tool, people who want a kind of curl like thing that has a pretty coloured output, should be happy with a reasonable, opinionated, default value.

You suggested 50, I think 10 is reasonable. Please pick one and make it a constant.

Thank you.

**Comment by jrozner** [comment]
I understand the sane default, which currently exists and changing to 10 is fine (50 was only chosen because it was the default in cURL), but why remove functionality that offers no real additional complexity? The python version allows passing virtually all flags to cURL not part of the the tool which isn't possible with implementing this entirely in Go

**Comment by davecheney** [comment]
> but why remove functionality that offers no real additional complexity

For the same rational as expressed here, https://github.com/davecheney/httpstat/issues/46#issuecomment-249411745

**Comment by ble** [comment]
I understand the desire to keep the interface simple; even though changing from a fixed default to an argument is could literally be a 1-line addition of code (`flag.UintVar(&maxRedirects, "max-redirs", 10, "maximum number of redirects to follow if -L is set"` e.g.), "one more argument" is a lot more complexity than "one more line of code".

**Comment by davecheney** [comment]
@ble I said no. I think the rational for wanting to limit the number of flags supported is reasonable.

File: main.go [hunk 1]
```diff
@@ -105,6 +110,11 @@ func headerKeyValue(h string) (string, string) {
 // visit visits a url and times the interaction.
 // If the response is a 30x, visit follows the redirect.
 func visit(url *url.URL) {
+	if redirectsFollowed > maxRedirects {
```
Comment (davecheney) [inline main.go]: Could you please move this logic down into the follow redirect code around line 257

**Comment by davecheney** [comment]
Thank you. I'll fix the merge conflicts and merge this now.

**Pull request merged**